src/vigenere/Vigenere.cc

```diff
@@ -0,0 +1,48 @@
+#include <iostream>
+#include "Vigenere.h"
+
+using namespace easy::encryption;
+
+int Vigenere::index(char c) {
+    for (int i = 0; i < VIGENERE_AVAILABLE_CHARS_SIZE; i++)
+        if (VIGENERE_AVAILABLE_CHARS[i] == c)
+            return i;
+    return -1;
+}
+
+std::string Vigenere::extend_key(const std::string &string, const std::string &key) {
+    unsigned long long int stringLen = string.size(), keyLen = key.size();
+    std::string newKey(stringLen, 0x0);
+
+    // Generating new key
+    for (int i = 0, j = 0; i < stringLen; ++i, ++j) {
+        if (j == keyLen) j = 0;
+        newKey[i] = key[j];
+    }
+
+    return newKey;
+}
+
+std::string Vigenere::encrypt(const std::string &decryptedString, const std::string &key) {
+    unsigned long long int decryptedStringLen = decryptedString.size();
+    std::string newKey = extend_key(decryptedString, key);
+    std::string encryptedString(decryptedStringLen, 0x0);
+
+    // Encryption
+    for (int i = 0; i < decryptedStringLen; ++i)
+        encryptedString[i] = isalnum(decryptedString[i]) || decryptedString[i] == 0x20 ? VIGENERE_AVAILABLE_CHARS[((index(decryptedString[i]) + index(newKey[i])) % VIGENERE_AVAILABLE_CHARS_SIZE)] : decryptedString[i];
+
+    return encryptedString;
+}
+
+std::string Vigenere::decrypt(const std::string &encryptedString, const std::string &key) {
+    unsigned long long int encryptedStringLen = encryptedString.size();
+    std::string newKey = extend_key(encryptedString, key);
+    std::string decryptedString(encryptedStringLen, 0x0);
+
+    // Decryption
+    for (int i = 0; i < encryptedStringLen; ++i)
+        decryptedString[i] = isalnum(encryptedString[i]) || encryptedString[i] == 0x20 ? VIGENERE_AVAILABLE_CHARS[(((index(encryptedString[i]) - index(newKey[i])) + VIGENERE_AVAILABLE_CHARS_SIZE) % VIGENERE_AVAILABLE_CHARS_SIZE)] : encryptedString[i];
+
+    return decryptedString;
+}
```

Comment:
```suggestion
std::string Vigenere::encrypt(const std::string &decryptedString, const std::string &key) {
    unsigned long long int decryptedStringLen = decryptedString.size();
    std::string newKey = extend_key(decryptedString, key);
    std::string encryptedString(decryptedStringLen, 0x0);

    // Encryption
    for (int i = 0; i < decryptedStringLen; ++i)
        encryptedString[i] = VIGENERE_AVAILABLE_CHARS[((index(decryptedString[i]) + index(newKey[i])) % VIGENERE_AVAILABLE_CHARS_SIZE)];

    return encryptedString;
}

std::string Vigenere::decrypt(const std::string &encryptedString, const std::string &key) {
    unsigned long long int encryptedStringLen = encryptedString.size();
    std::string newKey = extend_key(encryptedString, key);
    std::string decryptedString(encryptedStringLen, 0x0);

    // Decryption
    for (int i = 0; i < encryptedStringLen; ++i)
        decryptedString[i] = VIGENERE_AVAILABLE_CHARS[(((index(encryptedString[i]) - index(newKey[i])) + VIGENERE_AVAILABLE_CHARS_SIZE) % VIGENERE_AVAILABLE_CHARS_SIZE)];

    return decryptedString;
}
```
If we make these line changes the cryptography already almost matches, it just leaves a few extra characters at the end in the decryption.

![image](https://user-images.githubusercontent.com/38193239/116786048-b2686580-aa94-11eb-9fcf-05757f6a5a99.png)